pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -220,61 +218,53 @@ func (pl *PodTopologySpread) calPreFilterState(pod *v1.Pod) (*preFilterState, er
 	s := preFilterState{
 		Constraints:          constraints,
 		TpKeyToCriticalPaths: make(map[string]*criticalPaths, len(constraints)),
-		TpPairToMatchNum:     make(map[topologyPair]*int32, sizeHeuristic(len(allNodes), constraints)),
+		TpPairToMatchNum:     make(map[topologyPair]int32, sizeHeuristic(len(allNodes), constraints)),
 	}
 
-	// Nodes that pass nodeAffinity check and carry all required topology keys will be
-	// stored in `filteredNodes`, and be looped later to calculate preFilterState.
-	var filteredNodes []*framework.NodeInfo
 	requiredSchedulingTerm := nodeaffinity.GetRequiredNodeAffinity(pod)
-	for _, n := range allNodes {
-		node := n.Node()
+	tpCountsByNode := make([]map[topologyPair]int32, len(allNodes))
```

Comment:
Aha, we also used this "space-for-time" trick somewhere else.

Reply:
yup, in pod affinities https://github.com/kubernetes/kubernetes/blob/f03c06ad572e19f31d6b65090ca6eee63a28dbc1/pkg/scheduler/framework/plugins/interpodaffinity/scoring.go#L185 and https://github.com/kubernetes/kubernetes/blob/f03c06ad572e19f31d6b65090ca6eee63a28dbc1/pkg/scheduler/framework/plugins/interpodaffinity/filtering.go#L158